unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,235 @@
+                                 "Server will continue processing "
+                                 "your request on the background "
+                                 "and will notify you as soon as the "
+                                 "registration has been completed."):
+            redis_client_auto_register.register(is_blocking=True)
+        mock_wait_until_timeout_blocking.assert_called()
+
+    @staticmethod
+    def test_register_self_active_true_throws_exception(redis_client_auto_register):
+        """Check whether if is active is true throws exception."""
+        with pytest.raises(RedisAPIException,
+                           match="API is already registered to the market."):
+            redis_client_auto_register.is_active = True
+            redis_client_auto_register.register(is_blocking=False)
+
+    @staticmethod
+    @patch("uuid.uuid4", return_value="some-transaction-uuid")
+    def test_unregister(mock_uuid, redis_client_auto_register):
+        """Check whether the redis.db.publish is called with correct arguments."""
+        redis_client_auto_register.is_active = True
+        redis_client_auto_register.unregister(is_blocking=False)
+        data = {"name": AREA_ID, "transaction_id": mock_uuid()}
+        redis_client_auto_register.redis_db.publish.assert_called_once_with(
+            f"{AREA_ID}/unregister_participant",
+            json.dumps(data))
+
+    @staticmethod
+    def test_unregister_is_active_false_throws_exception(redis_client_auto_register):
+        """Check if is active set to false throws exception when user is already unregistered."""
+        with pytest.raises(RedisAPIException,
+                           match="API is already unregistered from the market."):
+            redis_client_auto_register.is_active = False
+            redis_client_auto_register.unregister(is_blocking=True)
+
+    @staticmethod
+    @patch("d3a_api_client.redis_client_base.wait_until_timeout_blocking",
+           side_effect=AssertionError)
+    def test_unregister_is_active_true_throws_exception(
+            mock_wait_until_timeout_blocking,
+            redis_client_auto_register):
+        """Check if is active set to true throws exception."""
+        with pytest.raises(RedisAPIException,
+                           match="API unregister process timed out. "
+                                 "Server will continue processing "
+                                 "your request on the background and "
+                                 "will notify you as soon as "
+                                 "the unregistration has been completed."):
+            redis_client_auto_register.is_active = True
+            redis_client_auto_register.unregister(is_blocking=True)
+        mock_wait_until_timeout_blocking.assert_called()
+
+    @staticmethod
+    @patch("uuid.uuid4", return_value="some-transaction-uuid")
+    def test_on_register(mock_uuid, redis_client_auto_register):
```

Comment:
Nit, In case you're testing the private _on_register method, there are three extra lines that can be tested

1. The mocked call to `_check_buffer_message_matching_command_and_id` method
2. the side effect (logging)
3. the call to the on_register method

Reply:
1. I would actually not assert the mocked call, since this is a private method. I would rather create a new test to check whether the exception is raised inside of it. However, that method has already been unit-tested, so if this is not compelling for you I might let it go. If you'd rather have it, I can definitely work on it.
2. Do you think we should test logging calls? We kept it out of scope by choice, but I can add it.
3. Done!

Thank you!